python/pyarrow/tests/test_udf.py

```diff
@@ -228,11 +253,11 @@ def check_scalar_function(func_fixture,
         if all_scalar:
             batch_length = 1
 
-    expected_output = function(mock_scalar_udf_context(batch_length), *inputs)
     func = pc.get_function(name)
     assert func.name == name
 
     result = pc.call_function(name, inputs, length=batch_length)
+    expected_output = function(mock_scalar_udf_context(batch_length), *inputs)
```

Comment:
Move this down to compute "result" first then "expected", which is how normally this is done in test